docs/assignment3.md

```diff
@@ -1 +1,205 @@
+
+The **Main** class is where the execution of the game happens. It prompts the user to input a JSON file and then reads in the JSON file to execute the game from the file. The *startRoom* variable is a JSONObject indicating the room that the game starts in, and *endRoom* indicates the room that the game ends in. *RoomMap* and *itemMap* are hashmaps indicating which rooms contain which items, based on the information in the JSON file. *PlayerName* is simply a string containing the player's name, which comes from the user when the game prompts them to input their name. The Main class is able to access the Room class and TimeLimit because the Main class needs to know where the player is and whether they have won yet or not. 
+
+The **Inventory** class holds all of the information about a player's inventory. The *items* attribute is an ArrayList of every item that is in the current inventory. The *getInventory* method returns the player's current inventory, i.e. what items they are currently holding, while *addToInventory* adds an item to this list. *PrintInventory* simply prints the name of each item in the items list, so a user can see what items they can perform actions on. *InventoryIsEmpty* checks whether the inventory has items in it, and *isInInventory* checks whether a specific item is in the inventory. This class is associated with the Player class because the player needs access to their inventory, and it is also associated with the Item class because the inventory holds items, and therefore needs access to them. 
+
+The **ChatClient** class is what allows a player to communicate with other players through a chat room. The *serverAddress* holds the server IP that will be used. *TextField*, *messageArea*, and *frame* are attributes that are used to build the window that the chat room is in. *In* and *out* represent the text coming in and out of the chat room. 
+
+In our original class diagram, we did not have an Inventory class. This led to the methods such as getInventory and printInventory being scattered across multiple classes, making things harder to understand and use. Adding an extra class for the inventory made more sense, since there are a lot more necessary methods involving the inventory than we thought. We also added a ChatClient class, where we implemented our chat room feature for our game. Now, our game supports the command "chat," and it will open a chat room on your screen, which you can then talk to other players on. In addition, we added association names to the diagram to make the arrows easier to follow. We also added an abstract class to represent the JSON file that the game reads in. 
+
```

Comment:
Great changes overall, a couple of notes:
-JSON file shouldn't be abstract
-This stems from the decorator to some extend I think but the way you structured your player as an interface isn't very intuitive. You would want Player to provide some core functionality as an abstract class to all possible future modified iterations of Player(such as the time-decorated player)  like getInventory or getName, so you would only need to extend additional functionality instead of re-writing everything.

Reply:
Very descriptive overall but I agree there is no need for the json file to be abstract.